tests/build_test.sh

```diff
@@ -53,8 +64,189 @@ tearDown()
   }
 }
 
+function test_parse_build_config_success_exit_code()
+{
+  build_config=()
+
+  parse_configuration "$KW_BUILD_CONFIG_SAMPLE" build_config
+  assertTrue 'kw failed to load a regular config file' "[ 0 -eq $? ]"
+}
+
+# Test if etc/kworkflow_template.config contains all the expected settings
+function test_parse_build_config_standard_config()
+{
+  build_config=()
+
+  declare -A expected_build_configurations=(
+    [arch]='x86_64'
+    [cpu_scaling_factor]='100'
+    [kernel_img_name]='bzImage'
+    [menu_config]='nconfig'
+    [doc_type]='htmldocs'
+  )
+
+  parse_configuration "$TMP_DIR/build.config" build_config
+  assertConfigurations build_config expected_build_configurations "$LINENO"
+
+  true # Reset return value
+}
+
+function assertConfigurations()
+{
+  declare -n configurations_ref=$1
+  declare -n expected_configurations_ref=$2
+  local lineno=${3:-LINENO}
```

Comment:
I think `LINENO` here will not work as expected; basically, it will point to this line instead of where `assertConfigurations` was invoked.

Reply:
I think it might actually work that way because `lineno=LINENO` but then you'd have to replace it twice when you use it, like `${!lineno}` (like a double pointer).